src/Package/Impl/ProjectSystem/IVsHierarchyExtensions.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using Microsoft.VisualStudio.ProjectSystem;
+using Microsoft.VisualStudio.ProjectSystem.Designers;
+using Microsoft.VisualStudio.Shell.Interop;
+
+namespace Microsoft.VisualStudio.R.Package.ProjectSystem {
+    internal static class IVsHierarchyExtensions {
+        /// <summary>
+        /// Returns EnvDTE.Project object for the hierarchy
+        /// </summary>
+        public static EnvDTE.Project GetDTEProject(this IVsHierarchy hierarchy) {
+            //UIThreadHelper.VerifyOnUIThread();
```

Comment:
CoreShellExtensions.AssertIsOnMainThread

---
In reply to: [63072611](https://github.com/Microsoft/RTVS/pull/1762#discussion_r63072611) [](ancestors = 63072611)